avalon/maya/pipeline.py

```diff
@@ -522,6 +522,9 @@ def remove(container):
     fname = cmds.referenceQuery(reference_node, filename=True)
     cmds.file(fname, removeReference=True)
 
+    if cmds.objExists(node):
```

Comment:
Ah, it looks like Maya is being smart and automatically deletes it for us under some circumstances.

Ok, in that case the only critique would be to delete first, ask later, in preparation for the day Maya becomes multi-threaded and could create data inbetween two subsequent commands, like a filesystem can.

```python
try:
  cmds.remove(..)
except TheError:
  pass
```